tidy3d/components/geometry/polyslab.py

```diff
@@ -1520,6 +1521,84 @@ def normalize_vect(arr: np.ndarray) -> np.ndarray:
         norm = np.where(norm == 0, 1, norm)
         return arr / norm
 
+    def translated(self, x: float, y: float, z: float) -> PolySlab:
+        """Return a translated copy of this geometry.
+
+        Parameters
+        ----------
+        x : float
+            Translation along x.
+        y : float
+            Translation along y.
+        z : float
+            Translation along z.
+
+        Returns
+        -------
+        :class:`PolySlab`
+            Translated copy of this ``PolySlab``.
+        """
+
+        t_normal, t_plane = self.pop_axis((x, y, z), axis=self.axis)
+        translated_vertices = np.array(self.vertices) + np.array(t_plane)[None, :]
+        translated_slab_bounds = (self.slab_bounds[0] + t_normal, self.slab_bounds[1] + t_normal)
+        return self.updated_copy(vertices=translated_vertices, slab_bounds=translated_slab_bounds)
+
+    def scaled(self, x: float = 1.0, y: float = 1.0, z: float = 1.0) -> PolySlab:
+        """Return a scaled copy of this geometry.
+
+        Parameters
+        ----------
+        x : float = 1.0
+            Scaling factor along x.
+        y : float = 1.0
+            Scaling factor along y.
+        z : float = 1.0
+            Scaling factor along z.
+
+        Returns
+        -------
+        :class:`Geometry`
+            Scaled copy of this geometry.
+        """
+        scale_normal, scale_in_plane = self.pop_axis((x, y, z), axis=self.axis)
+        scaled_vertices = self.vertices * np.array(scale_in_plane)
+        print(self.vertices[:5, :])
```

Comment:
Don't forget to remove the debug prints before merging.

Reply:
Ah thanks.